demo/src/asset-working-benchmark.ts

```diff
@@ -0,0 +1,245 @@
+  Cord.ConfigService.set({ submitTxResolveOn: Cord.Chain.IS_READY });
+  //Cord.ConfigService.set({ submitTxResolveOn: Cord.Chain.IS_IN_BLOCK});
+  //Cord.ConfigService.set({ submitTxResolveOn: Cord.Chain.IS_FINALIZED });
+  console.log = originalConsoleLog;
+
+  console.log(`\n❄️  Create On-Chain Assets  `);
+  let assetProperties: Cord.IAssetProperties = {
+    assetType: Cord.AssetTypeOf.art,
+    assetDesc: "Asset - " + Cord.Utils.UUID.generate(),
+    assetQty: 1000000,
+    assetValue: 10,
+    assetTag: "Tag - " + Cord.Utils.UUID.generate(),
+    assetMeta: "Meta - " + Cord.Utils.UUID.generate(),
+  };
+
+  const assetEntry = await Cord.Asset.buildFromAssetProperties(
+    assetProperties,
+    issuerDid.uri,
+    space.uri
+  );
+
+  console.dir(assetEntry, {
+    depth: null,
+    colors: true,
+  });
+
+  const extrinsic = await Cord.Asset.dispatchCreateToChain(
+    assetEntry,
+    issuerIdentity,
+    space.authorization,
+    async ({ data }) => ({
+      signature: issuerKeys.authentication.sign(data),
+      keyType: issuerKeys.authentication.type,
+    }),
+  )
+
+  console.log("✅ Asset created!");
+
+  console.log(`\n❄️  Transaction Benchmarking  `);
+
+  let tx_batch: any = [];
+  let startTxPrep = moment();
+
+  try {
+  for (let j = 0; j < (txCount / perBlock); j++) {
+     let tx_batch1: any = [];
+     for (let k = 0; k < perBlock; k++) {
+      const { account: holderIdentity } = createAccount();
+
+      const assetIssuance = await Cord.Asset.buildFromIssueProperties(
+        assetEntry.uri,
+        `did:cord:${holderIdentity.address}`,
+        1,
+        issuerDid.uri,
+        space.uri,
+      );
+      
+      const authorizationId: Cord.AuthorizationId = uriToIdentifier(space.authorization)
+  
+      const tx = await api.tx.asset.issue(
+        assetIssuance.entry,
+        assetIssuance.digest,
+        authorizationId
+      )
+
+      tx_batch1.push(tx);
+
+      process.stdout.write(
+        "  🔖  Preparing " +
+          (j * perBlock + 1 + k) +
+          " transactions took " +
+          moment.duration(moment().diff(startTxPrep)).as("seconds").toFixed(3) +
+          "s\r",
+      );
+      }
+      tx_batch[j] = tx_batch1;
+      }    
+    } catch (e: any) {
+      console.log(e.errorCode, "-", e.message);
+    }
+  console.log("\n");
+
+  //Cord.ConfigService.set({ submitTxResolveOn: Cord.Chain.IS_FINALIZED });
+  console.log = originalConsoleLog;
+  let batchAncStartTime = moment();
+
+  let promises = [];
+  for (let j = 0; j < tx_batch.length; j++) {
+  try {
+      let extSignCallback: Cord.SignExtrinsicCallback = async ({ data }) => ({
+            signature: issuerKeys.authentication.sign(data),
+            keyType: issuerKeys.authentication.type,
+      })
+     
+      /* Use nonce when authorizing tx in batch(loop) */
+     const authorizedBatch = await Cord.Did.authorizeBatch({
+      batchFunction: api.tx.utility.batchAll,
+      did: issuerDid.uri,
+      nonce: new BN(j+3),
```

Comment:
This is needed mainly because of async submission. In a case where only 1 utility.batchAll() is called per block, we wouldn't need nonce calculations. 

Leaving comment here for others to refer in future.